Gateway/Request/AuthorizationRequest.php

```diff
@@ -27,24 +25,36 @@ public function build(array $buildSubject)
             throw new \InvalidArgumentException('Payment data object should be provided');
         }
 
-        /** @var PaymentDataObjectInterface $payment */
-
         $paymentDO = $buildSubject['payment'];
         $amount = $this->formatPrice($buildSubject['amount']);
         $order = $paymentDO->getOrder();
         $payment = $paymentDO->getPayment();
-
+        $cc_exp_month = $payment->getAdditionalInformation("cc_exp_month");
+        $cc_exp_year = $payment->getAdditionalInformation("cc_exp_year");
         return [
             'xAmount' => $amount,
-            'xExp' => sprintf('%02d%02d', $payment->getAdditionalInformation("cc_exp_month"), substr($payment->getAdditionalInformation("cc_exp_year"), -2)),
+            'xExp' => sprintf('%02d%02d', $cc_exp_month, substr($cc_exp_year, -2)),
             'xCVV' => $payment->getAdditionalInformation("xCVV"),
             'xCommand' => 'cc:authonly',
             'xInvoice' => $order->getOrderIncrementId(),
             'xCurrency' => $order->getCurrencyCode(),
             'xCardNum' => $payment->getAdditionalInformation("xCardNum"),
             // always true; order number is incremented on every attempt so invoice is always different
             'xIgnoreInvoice' => true,
-            'xTimeoutSeconds' => 55
+            'xTimeoutSeconds' => 55,
+            'xAllowDuplicate' => true
```

Comment:
Was this intentional?

Reply:
It throws error when order place with same grand total with same time.
**Error**: Duplicate Transaction in payment.log

Reply:
Can you discuss with Y. Brodie? We don't want to lose duplicate protections.

Reply:
@mstein-ck , 
I removed **'xAllowDuplicate' => true** from all requests now.
Commit url: [65d858b](https://github.com/cardknox/magento2_cardknox/pull/27/commits/65d858bd4e28ffb4e06471b6760f1c205e2698e4)